File: Javascript/HighArray.js
function binarySearch(inputArray, searchKey) {
    if(!inputArray || inputArray.length == 0 || !searchKey) {
        return -1;
    }
    var len = inputArray.length;
    var upperBound = len - 1;
    var lowerBound = 0;
    while(true) {
        var middle = Math.round((lowerBound + upperBound)/2);
        if(searchKey === inputArray[middle]) {
            return middle;
        } else if(lowerBound > upperBound) {
            return -1;
        } else {
            if(searchKey < inputArray[middle]) {
                upperBound = middle-1;
            } else if (searchKey > inputArray[middle]) {
                lowerBound = middle + 1;
            }
        }
    }
}

console.log("key found at index : ")
console.log(binarySearch([1,3,4,6,8,34,56,89], 2));

File: src/HighArray.java
import java.util.ArrayList;

public class HighArray {
    public int find (int[] input, long value) {
        if(input.length <= 0) {
            return -1;
        }
        int length = input.length;
        int lowerBound = 0;
        int upperBound = length - 1;
        while (true) {
            int middle = (lowerBound + upperBound) / 2;
            if (value == input[middle]) {
                return middle;
            } else if (value < input[middle]) {
                upperBound = middle - 1;
            } else if (value > input[middle]) {
                lowerBound = middle + 1;
            }
            if (lowerBound > upperBound) {
                return -1;
            }
        }
    }

    public void insert(int[] input, int newVal) {

        
    }
}
